proto/interchain_security/ccv/provider/v1/tx.proto

```diff
@@ -210,3 +229,97 @@ message MsgChangeRewardDenoms {
 
 // MsgChangeRewardDenomsResponse defines response type for MsgChangeRewardDenoms messages
 message MsgChangeRewardDenomsResponse {}
+
+message MsgOptIn {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+  option (cosmos.msg.v1.signer) = "signer";
+  // the chain id of the consumer chain to opt in to
+  string chain_id = 1;
+  // the validator address on the provider
+  string provider_addr = 2 [ (gogoproto.moretags) = "yaml:\"address\"" ];
+  // (optional) The consensus public key to use on the consumer in json string format corresponding to proto-any,
+  // for example `{"@type":"/cosmos.crypto.ed25519.PubKey","key":"Ui5Gf1+mtWUdH8u3xlmzdKID+F3PK0sfXZ73GZ6q6is="}`.
+  // This field is optional and can remain empty (i.e., `consumer_key = ""`). A validator can always change the
+  // consumer public key at a later stage by issuing a `MsgAssignConsumerKey` message.
+  string consumer_key = 3;
+  // signer address
+  string signer = 4 [(cosmos_proto.scalar) = "cosmos.AddressString"];
```

Comment:
I guess `signer` here is due to SKD v0.50?
Why is it called `authority` in some messages but `signer` here?
Also, shouldn't `MsgSetConsumerCommissionRate` have a `signer`?

Reply:
IIUC `signer` is for normal msg and `authority` for proposals.